frontstage/views/register/confirm_organisation_survey.py

```diff
@@ -20,19 +19,31 @@ def register_confirm_organisation_survey():
     cryptographer = Cryptographer()
     encrypted_enrolment_code = request.args.get('encrypted_enrolment_code', None)
     enrolment_code = cryptographer.decrypt(encrypted_enrolment_code.encode()).decode()
-
-    logger.info('Attempting to retrieve data for confirm organisation/survey page')
-    response = api_call('POST', app.config['CONFIRM_ORGANISATION_SURVEY'],
-                        json={'enrolment_code': enrolment_code})
-
-    if response.status_code != 200:
-        logger.error('Failed to retrieve data for confirm organisation/survey page')
-        raise ApiError(response)
-
-    response_json = json.loads(response.text)
-    logger.info('Successfully retrieved data for confirm organisation/survey page')
+    # Validate enrolment code before retrieving organisation data
+    iac_controller.validate_enrolment_code(enrolment_code)
+
+    logger.info('Attempting to retrieve data for confirm organisation/survey page',
+                enrolment_code=enrolment_code)
+    try:
+        # Get organisation name
```

Comment:
Try blocks should be limited in their scope to isolate the error succinctly. This block has too much logic in it IMHO.

Reply:
I'm also a fan of limiting a try/except's scope but any one of the function calls within could raise an `ApiError` (when using one of the other services), so rather than blocking out each call I thought it made sense to capture and fail hard at any point during the confirmation process.